proposed/http-middleware/middleware.md

```diff
@@ -31,12 +31,11 @@ and the creation of a resulting HTTP Response, as defined by PSR-7.
 
 Middleware using this standard MUST implement the following interface:
 
-- `Psr\Http\ServerMiddleware\MiddlewareInterface`
+- `Psr\Http\Server\ServerMiddlewareInterface`
 
-Middleware dispatching systems using this standard MUST implement the following
-interface:
+Handlers using this standard MUST implement the following interface:
 
-- `Psr\Http\ServerMiddleware\DelegateInterface`
+- `Psr\Http\Server\HandlerInterface`
```

Comment:
In reading the rest of the document, I'm wondering if this should be `RequestHandlerInterface` or `ServerRequestHandlerInterface` instead? That would make it far more clear _what_ it handles.